src/train.js

```diff
@@ -165,26 +176,28 @@ function empty(element) {
 }
 
 function updateProgress(ratio, bestSolution, bestCost, successesOrFailures) {
-    document.getElementById('progress').setAttribute('value', ratio);
-
-    // Update best coeffs and accuracy:
-    const coeffsDiv = document.getElementById('coeffs');
-    const accuracyDiv = document.getElementById('accuracy');
-    Array.prototype.forEach.call(document.getElementsByClassName('output'),
-                                 e => e.classList.remove('output'));
-    coeffsDiv.classList.remove('output');
-    empty(coeffsDiv);
-    empty(accuracyDiv);
-    coeffsDiv.appendChild(document.createTextNode(`[${bestSolution}]`));
-    accuracyDiv.appendChild(document.createTextNode(`${((1 - bestCost) * 100).toFixed(1)}%`));
+    // Tick progress meter.
+    gProgressBar.setAttribute('value', ratio);
+
+    // Update best coeffs and accuracy.
+    gCoeffsDiv.firstChild.textContent = `[${bestSolution}]`;
+    gAccuracyDiv.firstChild.textContent = `${((1 - bestCost) * 100).toFixed(1)}%`;
+
     if (successesOrFailures) {
-        const goodBadDiv = document.getElementById('goodBad');
-        empty(goodBadDiv);
-        for (let [succeeded, name] of successesOrFailures) {
-            const newDiv = document.createElement('div');
-            newDiv.appendChild(document.createTextNode(name));
-            newDiv.setAttribute('class', succeeded ? 'good' : 'bad');
-            goodBadDiv.appendChild(newDiv);
+        if (gGoodBadDiv.childElementCount !== successesOrFailures.length) {
+            empty(gGoodBadDiv);
+            for (const _ of successesOrFailures) {
+                const div = document.createElement('div');
+                div.appendChild(document.createTextNode(''));
+                gGoodBadDiv.appendChild(div);
+            }
+        }
+
+        let div = gGoodBadDiv.firstElementChild;
+        for (const [succeeded, name] of successesOrFailures) {
```

Comment:
If you haven't already, make sure this works before merging. I generally find I have to use `let` with destructuring.